src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+        }
+        return getPickBlockDefault(state, target, world, pos, player);
+    }
+
+    public ItemStack getPickBlockDefault(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        return super.getPickBlock(state, target, world, pos, player);
+    }
+
+    private IMicroblockContainerTile brokenTile = null;
+    private boolean harvestingWrapper = false;
+
+    @Override
+    public final void harvestBlock(World world, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        brokenTile = te instanceof IMicroblockContainerTile ? (IMicroblockContainerTile) te : null;
+        harvestBlockDefault(world, player, pos, state, te, stack);
+        brokenTile = null;
+    }
+
+    public void harvestBlockDefault(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        super.harvestBlock(worldIn, player, pos, state, te, stack);
+    }
+
+    @Override
+    public final List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        if (harvestingWrapper) { return getDropsDefault(world, pos, state, fortune); }
+        IMicroblockContainerTile brokenTile = getMicroblockTile(world, pos);
+        if (brokenTile == null)
+        {
+            brokenTile = this.brokenTile;
+        }
+        if (brokenTile == null) { return getDropsDefault(world, pos, state, fortune); }
+        List<ItemStack> drops = new ArrayList<ItemStack>();
+        drops.addAll(getDropsDefault(world, pos, state, fortune));
+        return drops;
+    }
+
+    public List<ItemStack> getDropsDefault(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        return super.getDrops(world, pos, state, fortune);
+    }
+
+    @Override
+    public final boolean removedByPlayer(IBlockState state, World world, BlockPos pos, EntityPlayer player, boolean willHarvest)
+    {
+        RayTraceResult hit = reTraceAll(world, pos, player);
+        if (hit instanceof RayTraceResult)
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            if (tile != null)
+            {
+                tile.getMicroblockContainer().getPartContainer().harvest(player, (RayTraceResult) hit);
+            }
+            return false;
+        }
+        else
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            MultipartContainer container = tile != null ? tile.getMicroblockContainer().getPartContainer() : null;
+            if (container.getParts().isEmpty())
+            {
+                return removedByPlayerDefault(state, world, pos, player, willHarvest);
+            }
+            else
+            {
+                if (!removedByPlayerDefault(state, world, pos, player, willHarvest)) { return false; }
```

Comment:
Will do! I just ran all the code through a formatter that followed the guidelines cpw gave me and that was an option I wasn't quite sure how you wanted to handle, so I just did brackets because you like them. Them appearing on the same line... was more of a dep. If you don't mind one line stuff after ifs and fors without brackets, I'm more than happy to do that, since that's what I do. But I'd like you to confirm it first, I don't want to mess everything up... again...